cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/LaunchStackHandler.java

```diff
@@ -58,16 +59,20 @@ public void accept(Event<LaunchStackRequest> launchStackRequestEvent) {
         LOGGER.debug("Received event: {}", launchStackRequestEvent);
         LaunchStackRequest request = launchStackRequestEvent.getData();
         CloudContext cloudContext = request.getCloudContext();
+        CloudStack cloudStack = request.getCloudStack();
         try {
             CloudConnector<Object> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
             AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, request.getCloudCredential());
-            List<CloudResourceStatus> resourceStatus = connector.resources().launch(ac, request.getCloudStack(), persistenceNotifier,
+            List<CloudResourceStatus> resourceStatus = connector.resources().launch(ac, cloudStack, persistenceNotifier,
                     request.getAdjustmentType(), request.getThreshold());
-            List<CloudResource> resources = ResourceLists.transform(resourceStatus);
-            PollTask<ResourcesStatePollerResult> task = statusCheckFactory.newPollResourcesStateTask(ac, resources, true);
-            ResourcesStatePollerResult statePollerResult = ResourcesStatePollerResults.build(cloudContext, resourceStatus);
-            if (!task.completed(statePollerResult)) {
-                statePollerResult = syncPollingScheduler.schedule(task);
+            ResourcesStatePollerResult statePollerResult = waitForResources(ac, resourceStatus, cloudContext);
```

Comment:
What exactly do you mean by this? Should I check to see if the status is equal to FAILED or DELETED?
And if so, should I just not do the load balancer stuff, or handle that differently?

Reply:
@sxxgrc The `ResourcesStatePollerResult` contains the state of all the resources created in the cloud. At this point in the process, that should be all the instances, network resources, etc. What Tamas means is that we should check the state of those resources, and if they're failed we shouldn't do the load balancer stuff, since the creation will ultimately fail anyway. I think that would look something like this:
```
ResourcesStatePollerResult statePollerResult = waitForResources(ac, resourceStatus, cloudContext);
if (statePollerResult.getResults().stream().noneMatch(result -> ResourceStatus.FAILED.equals(result.getStatus()))) {
    launchLoadBalancers(cloudStack, connector, ac, persistenceNotifier, cloudContext, statePollerResult);
} else {
    LOGGER.error("Some cloud resources failed creation. Load balancers will not be created.");
}
LaunchStackResult result = ResourcesStatePollerResults.transformToLaunchStackResult(request, statePollerResult);
```
This isn't really something you've introduced. It should have been done earlier, but wasn't.

Reply:
@sxxgrc as @hreeve-cloudera mentioned can we get that additional condition? I can imagine that condition in the new `launchLoadBalancers` method too.

Reply:
Okay I added this condition to the launchLoadBalancers method. Let me know if y'all need anything else.

Reply:
By the way, the resulting LaunchStackResult object will contain the statuses for both the initial datalake instance launch and the loadbalancer launch, so that it will be aware of any failures across both launches and that will be handled by the general launch failure handling done later in the flow.